### ISSUE TITLE
toco error when converting tensorflow (Alexnet caffe model converted to tensorflow using MMdnn) to tflite

### ISSUE THREAD

**Issue by rohith-k-sturfee** [tensorflow/tensorflow#19949]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS High Sierra
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7.1
- **Python version**: 2.7.15
- **Bazel version (if compiling from source)**: 
- **GCC/Compiler version (if compiling from source)**: Apple LLVM version 9.1.0 (clang-902.0.39.2)
- **CUDA/cuDNN version**: No
- **GPU model and memory**:No 
- **Exact command to reproduce**: toco --input_file='inferencemodel/inference_inorout.pb' --input_format=TENSORFLOW_GRAPHDEF --output_format=TFLITE --output_file='tflite/model.tflite' --inference_type = FLOAT --input_data_type = FLOAT --input_array='input' --output_array='output' --input_shape=1,227,227,3

Hi,

I have a caffe model (Retrained Alexnet), converted this into tensorflow model using following commands. (https://github.com/Microsoft/MMdnn/issues/10)

```python -m mmdnn.conversion._script.convertToIR -f caffe -d kit_imagenet -n examples/caffe/models/bvlc_alexnet.prototxt -w examples/caffe/models/bvlc_alexnet.caffemodel```

```python -m mmdnn.conversion._script.IRToCode -f tensorflow --IRModelPath kit_imagenet.pb --dstModelPath kit_imagenet.py -w kit_imagenet.npy``` 

It generated python file kit_imagenet.py and numpy weights kit_imagenet.npy.
Kit_imagenet.py looks like this
```
import tensorflow as tf
__weights_dict = dict()
is_train = False

def load_weights(weight_file):
    import numpy as np
    if weight_file == None:
        return
    try:
        weights_dict = np.load(weight_file).item()
    except:
        weights_dict = np.load(weight_file, encoding='bytes').item()
    return weights_dict

def KitModel(weight_file = None):
    global __weights_dict
    __weights_dict = load_weights(weight_file)

    data            = tf.placeholder(tf.float32, shape = (None, 227, 227, 3), name = 'input')
    conv1_pad       = tf.pad(data, paddings = [[0L, 0L], [0L, 1L], [0L, 1L], [0L, 0L]])
    conv1           = convolution(conv1_pad, group=1, strides=[4, 4], padding='VALID', name='conv1')
    relu1           = tf.nn.relu(conv1, name = 'relu1')
    pool1_pad       = tf.pad(relu1, paddings = [[0L, 0L], [0L, 1L], [0L, 1L], [0L, 0L]], constant_values=float('-Inf'))
    pool1           = tf.nn.max_pool(pool1_pad, [1, 3, 3, 1], [1, 2, 2, 1], padding='VALID', name='pool1')
    norm1           = tf.nn.lrn(pool1, 2, alpha = 1.99999994948e-05, beta = 0.75, name = 'norm1')
    conv2_pad       = tf.pad(norm1, paddings = [[0L, 0L], [2L, 2L], [2L, 2L], [0L, 0L]])
    conv2           = convolution(conv2_pad, group=2, strides=[1, 1], padding='VALID', name='conv2')
    relu2           = tf.nn.relu(conv2, name = 'relu2')
    pool2_pad       = tf.pad(relu2, paddings = [[0L, 0L], [0L, 1L], [0L, 1L], [0L, 0L]], constant_values=float('-Inf'))
    pool2           = tf.nn.max_pool(pool2_pad, [1, 3, 3, 1], [1, 2, 2, 1], padding='VALID', name='pool2')
    norm2           = tf.nn.lrn(pool2, 2, alpha = 1.99999994948e-05, beta = 0.75, name = 'norm2')
    conv3_pad       = tf.pad(norm2, paddings = [[0L, 0L], [1L, 1L], [1L, 1L], [0L, 0L]])
    conv3           = convolution(conv3_pad, group=1, strides=[1, 1], padding='VALID', name='conv3')
    relu3           = tf.nn.relu(conv3, name = 'relu3')
    conv4_pad       = tf.pad(relu3, paddings = [[0L, 0L], [1L, 1L], [1L, 1L], [0L, 0L]])
    conv4           = convolution(conv4_pad, group=2, strides=[1, 1], padding='VALID', name='conv4')
    relu4           = tf.nn.relu(conv4, name = 'relu4')
    conv5_pad       = tf.pad(relu4, paddings = [[0L, 0L], [1L, 1L], [1L, 1L], [0L, 0L]])
    conv5           = convolution(conv5_pad, group=2, strides=[1, 1], padding='VALID', name='conv5')
    relu5           = tf.nn.relu(conv5, name = 'relu5')
    pool5_pad       = tf.pad(relu5, paddings = [[0L, 0L], [0L, 1L], [0L, 1L], [0L, 0L]], constant_values=float('-Inf'))
    pool5           = tf.nn.max_pool(pool5_pad, [1, 3, 3, 1], [1, 2, 2, 1], padding='VALID', name='pool5')
    fc6_0           = tf.contrib.layers.flatten(pool5)
    fc6_1           = tf.layers.dense(fc6_0, 4096, kernel_initializer = tf.constant_initializer(__weights_dict['fc6_1']['weights']), bias_initializer = tf.constant_initializer(__weights_dict['fc6_1']['bias']), use_bias = True)
    relu6           = tf.nn.relu(fc6_1, name = 'relu6')
    fc7_0           = tf.contrib.layers.flatten(relu6)
    fc7_1           = tf.layers.dense(fc7_0, 4096, kernel_initializer = tf.constant_initializer(__weights_dict['fc7_1']['weights']), bias_initializer = tf.constant_initializer(__weights_dict['fc7_1']['bias']), use_bias = True)
    relu7           = tf.nn.relu(fc7_1, name = 'relu7')
    fc8_indoor_outdoor_0 = tf.contrib.layers.flatten(relu7)
    fc8_indoor_outdoor_1 = tf.layers.dense(fc8_indoor_outdoor_0, 2, kernel_initializer = tf.constant_initializer(__weights_dict['fc8-indoor-outdoor_1']['weights']), bias_initializer = tf.constant_initializer(__weights_dict['fc8-indoor-outdoor_1']['bias']), use_bias = True)
    prob            = tf.nn.softmax(fc8_indoor_outdoor_1, name = 'output')
    return data, prob

def convolution(input, name, group, **kwargs):
    w = tf.Variable(__weights_dict[name]['weights'], trainable=is_train, name=name + "_weight")
    if group == 1:
        layer = tf.nn.convolution(input, w, **kwargs)
    else:
        weight_groups = tf.split(w, num_or_size_splits=group, axis=-1)
        xs = tf.split(input, num_or_size_splits=group, axis=-1)
        convolved = [tf.nn.convolution(x, weight, **kwargs) for
                    (x, weight) in zip(xs, weight_groups)]
        layer = tf.concat(convolved, axis=-1)

    if 'bias' in __weights_dict[name]:
        b = tf.Variable(__weights_dict[name]['bias'], trainable=is_train, name=name + "_bias")
        layer = layer + b
    return layer

```

I then wrote a script which uses kit_imagenet.py and numpy weights kit_imagenet.npy to create freezed model
```
import tensorflow as tf
from freeze_graph import freeze_graph # tensorflow comes up with a tool allowing freeze graph
from kit_imagenet import KitModel
from tensorflow.python.framework import graph_util
import os

data_node, prob_node = KitModel('kit_imagenet.npy')
sess = tf.InteractiveSession()
sess.run(tf.global_variables_initializer())

model_dir = './saved_model'
MODEL_NAME = "inorout"

saver = tf.train.Saver()
saver.save(sess, os.path.join(model_dir,MODEL_NAME), global_step=0, latest_filename='chkpt_state')

graph_def = tf.get_default_graph().as_graph_def()
output_graph = graph_util.convert_variables_to_constants(sess, graph_def, ['output'])
with tf.gfile.GFile(os.path.join(model_dir, MODEL_NAME + '.pb'), 'wb') as f:
    f.write(output_graph.SerializeToString())
```

After having freezed model, I converted to optimized version of protobuf using (tensorflow.python.tools.optimize_for_inference ). now i have 'inference_inorout.pb'.

Now I am trying to converted this optimized version of protobuf to tflite model, but it shows errors!

```

toco --input_file='inferencemodel/inference_inorout.pb' --input_format=TENSORFLOW_GRAPHDEF --output_format=TFLITE --output_file='tflite/model.tflite' --inference_type = FLOAT --input_data_type = FLOAT --input_array='input' --output_array='output' --input_shape=1,227,227,3
2018-06-12 10:53:52.151081: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1233] Converting unsupported operation: PadV2
2018-06-12 10:53:52.153847: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1233] Converting unsupported operation: PadV2
2018-06-12 10:53:52.161334: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1233] Converting unsupported operation: PadV2
2018-06-12 10:53:52.457117: F tensorflow/contrib/lite/toco/tooling_util.cc:842] Check failed: name.substr(colon_pos + 1).find_first_not_of("0123456789") == string::npos (1 vs. 18446744073709551615)Array name must only have digits after colon
Abort trap: 6 ```

**Comment by pillarpond** [comment]
I'm not sure that the issue is exactly same with my case.
But I saw same error msg and have made pull request #19226.
The request is not reviewed yet.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @michaelisard: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by anitha-v** [comment]
We believe this issue should be fixed in the nightly. PadV2 is now added and the 0123456789 first_of_ error should also be fixed. Please let us know if this works.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rohith-k-sturfee** [comment]
Still Error, I also used tflite_convert (TensorFlow 1.9) instead of toco.

**Comment by suharshs** [comment]
Which error remains? Did you use the new nightly build?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rohith-k-sturfee** [comment]
Now I used new nightly build. This error remains same:
```2018-09-10 09:00:50.268177: F tensorflow/contrib/lite/toco/tooling_util.cc:981] Check failed: name.substr(colon_pos + 1).find_first_not_of("0123456789") == string::npos (1 vs. 18446744073709551615)Array name must only have digits after colon```

**Comment by tofulawrence** [comment]
Can you run the graph just using TF core? It looks like the resulting TF graph has invalid tensor names.

**Comment by rohith-k-sturfee** [comment]
Following are the tensor names
`Placeholder`
`Placeholder_1`
`input`
`Pad/paddings`
`Pad`
`conv1_weight/initial_value`
`conv1_weight`
`conv1_weight/Assign`
`conv1_weight/read`
`convolution/dilation_rate`
`convolution`
`conv1_bias/initial_value`
`conv1_bias`
`conv1_bias/Assign`
`conv1_bias/read`
`add`
`relu1`
`PadV2/paddings`
`PadV2/constant_values`
`PadV2`
`pool1`
`norm1`
`Pad_1/paddings`
`Pad_1`
`conv2_weight/initial_value`
`conv2_weight`
`conv2_weight/Assign`
`conv2_weight/read`
`Const`
`split/split_dim`
`split`
`Const_1`
`split_1/split_dim`
`split_1`
`convolution_1/dilation_rate`
`convolution_1`
`convolution_2/dilation_rate`
`convolution_2`
`concat/axis`
`concat`
`conv2_bias/initial_value`
`conv2_bias`
`conv2_bias/Assign`
`conv2_bias/read`
`add_1`
`relu2`
`PadV2_1/paddings`
`PadV2_1/constant_values`
`PadV2_1`
`pool2`
`norm2`
`Pad_2/paddings`
`Pad_2`
`conv3_weight/initial_value`
`conv3_weight`
`conv3_weight/Assign`
`conv3_weight/read`
`convolution_3/dilation_rate`
`convolution_3`
`conv3_bias/initial_value`
`conv3_bias`
`conv3_bias/Assign`
`conv3_bias/read`
`add_2`
`relu3`
`Pad_3/paddings`
`Pad_3`
`conv4_weight/initial_value`
`conv4_weight`
`conv4_weight/Assign`
`conv4_weight/read`
`Const_2`
`split_2/split_dim`
`split_2`
`Const_3`
`split_3/split_dim`
`split_3`
`convolution_4/dilation_rate`
`convolution_4`
`convolution_5/dilation_rate`
`convolution_5`
`concat_1/axis`
`concat_1`
`conv4_bias/initial_value`
`conv4_bias`
`conv4_bias/Assign`
`conv4_bias/read`
`add_3`
`relu4`
`Pad_4/paddings`
`Pad_4`
`conv5_weight/initial_value`
`conv5_weight`
`conv5_weight/Assign`
`conv5_weight/read`
`Const_4`
`split_4/split_dim`
`split_4`
`Const_5`
`split_5/split_dim`
`split_5`
`convolution_6/dilation_rate`
`convolution_6`
`convolution_7/dilation_rate`
`convolution_7`
`concat_2/axis`
`concat_2`
`conv5_bias/initial_value`
`conv5_bias`
`conv5_bias/Assign`
`conv5_bias/read`
`add_4`
`relu5`
`PadV2_2/paddings`
`PadV2_2/constant_values`
`PadV2_2`
`pool5`
`Flatten/flatten/Shape`
`Flatten/flatten/strided_slice/stack`
`Flatten/flatten/strided_slice/stack_1`
`Flatten/flatten/strided_slice/stack_2`
`Flatten/flatten/strided_slice`
`Flatten/flatten/Reshape/shape/1`
`Flatten/flatten/Reshape/shape`
`Flatten/flatten/Reshape`
`dense/kernel/Initializer/Const`
`dense/kernel`
`dense/kernel/Assign`
`dense/kernel/read`
`dense/bias/Initializer/Const`
`dense/bias`
`dense/bias/Assign`
`dense/bias/read`
`dense/MatMul`
`dense/BiasAdd`
`relu6`
`Flatten_1/flatten/Shape`
`Flatten_1/flatten/strided_slice/stack`
`Flatten_1/flatten/strided_slice/stack_1`
`Flatten_1/flatten/strided_slice/stack_2`
`Flatten_1/flatten/strided_slice`
`Flatten_1/flatten/Reshape/shape/1`
`Flatten_1/flatten/Reshape/shape`
`Flatten_1/flatten/Reshape`
`dense_1/kernel/Initializer/Const`
`dense_1/kernel`
`dense_1/kernel/Assign`
`dense_1/kernel/read`
`dense_1/bias/Initializer/Const`
`dense_1/bias`
`dense_1/bias/Assign`
`dense_1/bias/read`
`dense_1/MatMul`
`dense_1/BiasAdd`
`relu7`
`Flatten_2/flatten/Shape`
`Flatten_2/flatten/strided_slice/stack`
`Flatten_2/flatten/strided_slice/stack_1`
`Flatten_2/flatten/strided_slice/stack_2`
`Flatten_2/flatten/strided_slice`
`Flatten_2/flatten/Reshape/shape/1`
`Flatten_2/flatten/Reshape/shape`
`Flatten_2/flatten/Reshape`
`dense_2/kernel/Initializer/Const`
`dense_2/kernel`
`dense_2/kernel/Assign`
`dense_2/kernel/read`
`dense_2/bias/Initializer/Const`
`dense_2/bias`
`dense_2/bias/Assign`
`dense_2/bias/read`
`dense_2/MatMul`
`dense_2/BiasAdd`
`output`
`init`

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suharshs: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suharshs: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Scarecrow0** [comment]
I met this problem too.

> 2018-10-25 16:36:27.728852: F tensorflow/contrib/lite/toco/tooling_util.cc:937] Check failed: name.substr(colon_pos + 1).find_first_not_of("0123456789") == string::npos (1 vs. 18446744073709551615)Array name must only have digits after colon\nAborted (core dumped)\n

None of the names of the tensors have a colon, so why it happened?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suharshs: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by andrehentz** [comment]
The error is being issued by: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/toco/tooling_util.cc#L1038

Unfortunate, it doesn't include enough information to pinpoint the troublesome name. An upcoming change will improve the error message. Could you try again in a few days?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suharshs: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by suharshs** [comment]
Closing due to inactivity.
